File: Node.js
class Node {
    constructor(inputStreet, inputlistNeighbors, inputisTarget) {
        this.street = inputStreet;
        this.listNeighbors = inputlistNeighbors || [];
        this.isTarget = inputisTarget || false;
        this.next = null;
    }

    setListNeighbors(listNeighbors) {
        this.listNeighbors = listNeighbors;
    }

    setisTarget(b) {
        this.isTarget = b;
    }
}

module.exports = Node; // Exporting the Graph2 class for use in other files

File: Node.java
package src.backend;

class Node{
        public String street;
        Node[] listNeighbors;
        boolean isTarget;
        Node next;

        public Node(String inputStreet, Node[] inputlistNeighbors, boolean inputisTarget){
            this.street = inputStreet;
            this.listNeighbors = inputlistNeighbors;
            this.isTarget = inputisTarget;
        }

        public void setListNeighbors(Node[] listNeighbors) {
            this.listNeighbors = listNeighbors;
        }

        public void printNeighbors(){
            for (int i = 0; i < this.listNeighbors.length; i++){
                System.out.println(this.listNeighbors[i].street);
            }
        }

        public void setisTarget(boolean b){
            this.isTarget = b;
        }

}